From: Christoph Hellwig <hch@lst.de>
Date: Fri, 9 Jun 2017 16:36:50 +0200
Subject: thermal: int340x_thermal: fix compile after the UUID API switch
Patch-mainline: v4.13-rc1
Git-commit: 87085ff2e90ecfa91f8bb0cb0ce19ea661bd6f83
References: FATE#324505

Fix the compile after the switch to the UUID API in commit f4c19ac9
("thermal: int340x_thermal: Switch to use new generic UUID API").

Signed-off-by: Christoph Hellwig <hch@lst.de>
Acked-by: Johannes Thumshirn <jthumshirn@suse.de>
---
 drivers/thermal/int340x_thermal/int3400_thermal.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/thermal/int340x_thermal/int3400_thermal.c
+++ b/drivers/thermal/int340x_thermal/int3400_thermal.c
@@ -23,7 +23,7 @@ enum int3400_thermal_uuid {
 	INT3400_THERMAL_MAXIMUM_UUID,
 };
 
-static const char *int3400_thermal_uuids[INT3400_THERMAL_MAXIMUM_UUID] = {
+static char *int3400_thermal_uuids[INT3400_THERMAL_MAXIMUM_UUID] = {
 	"42A441D6-AE6A-462b-A84B-4A8CE79027D3",
 	"3A95C389-E4B8-4629-A526-C52C88626BAE",
 	"97C68AE7-15FA-499c-B8C9-5DA81D606E0A",
@@ -144,7 +144,7 @@ static int int3400_thermal_get_uuids(str
 			guid_t guid;
 
 			guid_parse(int3400_thermal_uuids[j], &guid);
-			if (guid_equal(objb->buffer.pointer, &guid)) {
+			if (guid_equal((guid_t *)objb->buffer.pointer, &guid)) {
 				priv->uuid_bitmap |= (1 << j);
 				break;
 			}
